sound/soc/sof/ipc.c

```diff
@@ -784,6 +784,34 @@ int snd_sof_ipc_set_get_comp_data(struct snd_sof_control *scontrol,
 }
 EXPORT_SYMBOL(snd_sof_ipc_set_get_comp_data);
 
+/* Helper of sending IPC to enable the DSP cores with cores_mask */
+int snd_sof_ipc_core_enable(struct snd_sof_dev *sdev, u32 core_mask)
+{
+	struct sof_ipc_pm_core_config pm_core_config;
+	int err;
+
+	/* check if IPC message needed */
+	if (core_mask == sdev->enabled_cores_mask)
```

Comment:
since you aren't taking the mutex in the helper, you expect the caller to take it? Usually it's good to do that internally, I think. And again - if you find it possible to not include sending the IPC under the mutex, you have to do that internally here. Otherwise it should be documented, that lock x should be held when calling this